src/components/Heatmaps.tsx

```diff
@@ -59,46 +64,46 @@ const getAvailableProperties = (
   }
 };
 
-interface HeatmapProps {
-  points: SurveyPoint[];
-  dimensions: { width: number; height: number };
-  image: string;
-}
+export function Heatmaps() {
+  const { settings, updateSettings } = useSettings();
+
+  console.log(`opening Heatmaps: ${JSON.stringify(settings)}`);
 
-export const Heatmaps: React.FC<HeatmapProps> = ({
-  points,
-  dimensions,
-  image,
-}) => {
   const [heatmaps, setHeatmaps] = useState<{ [key: string]: string | null }>(
     {},
   );
   const [selectedHeatmap, setSelectedHeatmap] = useState<{
     src: string;
     alt: string;
   } | null>(null);
+
   const [selectedMetrics, setSelectedMetrics] = useState<MeasurementTestType[]>(
     ["signalStrength", "tcpDownload", "tcpUpload"],
   );
   const [selectedProperties, setSelectedProperties] = useState<
     (keyof IperfTestProperty)[]
   >(["bitsPerSecond"]);
+
   const [showSignalStrengthAsPercentage, setShowSignalStrengthAsPercentage] =
     useState(true);
 
-  const [heatmapConfig, setHeatmapConfig] = useState<HeatmapConfig>({
-    radiusDivider: 1,
-    maxOpacity: 0.7,
-    minOpacity: 0.2,
-    blur: 0.99,
-    gradient: {
-      0.0: "rgba(0, 0, 255, 0.6)",
-      0.25: "rgba(0, 255, 255, 0.6)",
-      0.5: "rgba(0, 255, 0, 0.6)",
-      0.75: "rgba(255, 255, 0, 0.6)",
-      1.0: "rgba(255, 0, 0, 0.6)",
-    },
-  });
+  // const r1 = calculateRadiusByDensity; // bad for small numbers of points
```

Comment:
THis is an exception where I welcome the comments and keeping commented code. Great, thanks.